### PR TITLE
Feat(icons): Do not remove colors from icons with `-colored` suffix #DS-990

### PR DESCRIPTION
**Pull request by crishpeen**
### Issue reference

DS-990

---

### Before submitting the PR, please make sure you do the following

- [x] Read the [Contributing Guidelines](https://github.com/lmc-eu/spirit-design-system/blob/main/CONTRIBUTING.md).
- [x] Follow the [PR Title/Commit Message Convention](https://github.com/lmc-eu/spirit-design-system/blob/main/CONTRIBUTING.md#commit-conventions).
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [x] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

**Review by pavelklibani — APPROVED** [review]

File: packages/icons/scripts/buildSvg.js [hunk 1]
```diff
@@ -17,7 +17,7 @@ const normalizeAndCopySvg = (srcDir, distDir) => {
         const svgPath = path.join(srcDir, svg);
         const svgDistPath = path.join(distDir, svg);
         const svgContent = fs.readFileSync(svgPath, 'utf8');
-        const svgContentFixed = svgContent.replace(/fill="#\w+"/g, 'fill="currentColor"');
+        const svgContentFixed = svg.endsWith('-colored.svg') ? svgContent : svgContent.replace(/fill="#\w+"/g, 'fill="currentColor"');
```
Comment (literat) [inline packages/icons/scripts/buildSvg.js:20]: I strongly suggest creating a function for this -> `const normalizeSvgColors = (svg, svgContent) => svg.endsWith('-colored.svg') ? svgContent : svgContent.replace(/fill="#\w+"/g, 'fill="currentColor"');`

`const svgContentNormalized = normalizeSvgColors(svg, svgContent)`.

In this case the `svgContentFixed` name is out-of-date ;-)

Comment (crishpeen) [reply packages/icons/scripts/buildSvg.js]: done

File: packages/icons/scripts/buildSvg.js [hunk 2]
```diff
@@ -5,6 +5,8 @@ const { filterSvgFiles } = require('./shared');
 const svgSrcDir = path.resolve(__dirname, `../src/svg`);
 const svgDistDir = path.resolve(__dirname, `../dist/svg`);
 
+const normalizeSvgColors = (svg, svgContent) => svg.endsWith('-colored.svg') ? svgContent : svgContent.replace(/fill="#\w+"/g, 'fill="currentColor"');
```
Comment (adamkudrna) [inline packages/icons/scripts/buildSvg.js:8]:
```suggestion
const normalizeSvgColors = (fileName, svgContent) => fileName.endsWith('-colored.svg') ? svgContent : svgContent.replace(/fill="#\w+"/g, 'fill="currentColor"');
```

?

**Review by adamkudrna — APPROVED** [review]
Thanks for the docs 👍🏻.

File: packages/icons/scripts/buildSvg.js [hunk 3]
```diff
@@ -17,7 +19,7 @@ const normalizeAndCopySvg = (srcDir, distDir) => {
         const svgPath = path.join(srcDir, svg);
         const svgDistPath = path.join(distDir, svg);
         const svgContent = fs.readFileSync(svgPath, 'utf8');
-        const svgContentFixed = svgContent.replace(/fill="#\w+"/g, 'fill="currentColor"');
+        const svgContentFixed = normalizeSvgColors(svg, svgContent);
```
Comment (literat) [inline packages/icons/scripts/buildSvg.js:22]: I do not like the `svgContentFixed` name. Can we use `svgContentNormalized` instead? Or just append the `replace` behind the call - `const svgSpriteContent = normalizeSvgColors(svg, svgContent).replace(...).replace(...)`?

Comment (crishpeen) [reply packages/icons/scripts/buildSvg.js:22]: We use it here too
```
fs.writeFileSync(path.join(svgDistPath), svgContentNormalized);
```
But I renamed it to `svgContentNormalized`

**Pull request merged**